mp/src/game/client/momentum/ui/HUD/hud_speedometer_data.cpp

```diff
@@ -0,0 +1,200 @@
+#include "cbase.h"
+
+#include "hud_speedometer_data.h"
+#include "filesystem.h"
+
+#include "tier0/memdbgon.h"
+
+#define SPEEDOMETER_FILENAME "cfg/speedometer.vdf"
+#define SPEEDOMETER_DEFAULT_FILENAME "cfg/speedometer_default.vdf"
+
+CON_COMMAND_F(mom_hud_speedometer_savecfg,
+              "Writes the current speedometer setup for the current gamemode to file.\n"
+              "Optionally takes in the gamemode to write to.\n",
+              FLAG_HUD_CVAR | FCVAR_CLIENTCMD_CAN_EXECUTE)
+{
+    if (args.ArgC() > 1)
+    {
+        g_pSpeedometerData->SaveGamemodeData(g_pGameModeSystem->GetGameMode(Q_atoi(args[1]))->GetType());
+    }
+    else
+    {
+        g_pSpeedometerData->SaveGamemodeData(g_pGameModeSystem->GetGameMode()->GetType());
+    }
+}
+
+CON_COMMAND_F(mom_hud_speedometer_loadcfg,
+              "Loads the speedometer setup for the current gamemode from file.\n"
+              "Optionally takes in the gamemode to load from.",
+              FLAG_HUD_CVAR | FCVAR_CLIENTCMD_CAN_EXECUTE)
+{
+    if (args.ArgC() > 1)
+    {
+        g_pSpeedometerData->LoadGamemodeData(g_pGameModeSystem->GetGameMode(Q_atoi(args[1]))->GetType());
```

Comment:
You should probably check the outcome of `Q_atoi(args[1])` and see if it's a valid value, otherwise we'd get a nasty crash here. Do this for both load and save.